[DOC] update to BEP process (#615)
bids-standard/bids-website: docs/extensions/process.md

# BEP process

Small contributions (typos, rephrasing of a description, adding a single new metadata field)

> can be just added as a [Pull Request on GitHub](https://github.com/bids-standard/bids-specification/pulls)

Reviewer: ```suggestion
can be proposed as a [Pull Request on GitHub](https://github.com/bids-standard/bids-specification/pulls)
```

Larger contributions that are expected to involve longer and more involved discussions

> may take the form of a BIDS extension proposal.

Reviewer: ```suggestion
may take the form of a BIDS extension proposal (BEP).
```

A BEP is a method of expanding the BIDS Specification to encompass new features or data types.

They are called BEPs because they are modeled after
[Python Extension Proposals](https://peps.python.org/pep-0001/#what-is-a-pep) (PEPs) as
they have been an effective community tool to change (either by updating or supplementing)

> the Python programming language.

Reviewer: ```suggestion
the [Python programming language](https://www.python.org/).
```

draft_bep --> draft_bep_review
draft_bep_review -- negative evaluation --> draft_bep
draft_bep_review -- positive evaluation --> proposed_bep
proposed_bep --> proposed_bep_review
proposed_bep_review -- negative evaluation --> proposed_bep
proposed_bep_review -- positive evaluation --> merged_bep
```

It may also be decided to close a BEP that is not actually necessary anymore
or is in fact beyond the scope of BIDS.

```mermaid
flowchart TD

draft_bep[Draft BEP]
draft_bep_review([Draft BEP review])
proposed_bep[Proposed BEP]
proposed_bep_review([Proposed BEP review])
merged_bep[Merged BEP]
closed_BEP[Closed BEP]

draft_bep --> closed_BEP
proposed_bep --> closed_BEP
draft_bep --> draft_bep_review
draft_bep_review -- negative evaluation --> draft_bep
draft_bep_review -- positive evaluation --> proposed_bep
proposed_bep --> proposed_bep_review
proposed_bep_review -- negative evaluation --> proposed_bep
proposed_bep_review -- positive evaluation --> merged_bep
```

### Deliverables

In the end a BEP should produce several deliverables
that will be evaluated during the `Proposed BEP review`.

```mermaid
flowchart TD

draft_bep[Draft BEP]
draft_bep_review([Draft BEP review])

> proposed_bep[Proposed BEP]

Reviewer: that's about PR right? so make it explicit ?
```suggestion
proposed_bep[Proposed BEP Pull Request]
```

PR_examples[REQUIRED

>             Pull request to examples repository]

Reviewer: could incorporate URL hyperlink, WDYT?

```suggestion
            Pull request to <a href='https://github.com/bids-standard/bids-examples'>examples repository</a>]
```

since images, should we shorten to 

```suggestion
            PR to <a href='https://github.com/bids-standard/bids-examples'>examples repository</a>]
```

with abbreviation in the text? After all we have BEP here too anyways

Author: not sure that mermaid handles link well within mkdocs: let's keep it for later

    in order to make sure
    that more consensus arises and more contributors are aware what is happening.**

>     <!-- TODO - create an issue template for BEPs -->

Reviewer: I love seeing you using `datalad`! 
Remember that it also has `-m` for `save` and `run` ;)

Reviewer: time to link to some llm to summarize a commit msg for you too? ;)

Author: PR are set for squash merge on this repo: I'd rather spend my effort on the global PR description than individual commit
